library/Network/IPFS/Local/Class.hs

```diff
@@ -1,25 +1,17 @@
-{-# LANGUAGE UndecidableInstances #-}
-
 module Network.IPFS.Local.Class
   ( MonadLocalIPFS
-  , withIPFSProc
   , ipfsRun
-  , getTimeout
   ) where
 
 import Network.IPFS.Prelude
 
 import qualified RIO.ByteString.Lazy as Lazy
 
-import           Network.IPFS.Types  as IPFS
-import           Network.IPFS.Internal.Process
+import           Network.IPFS.Types         as IPFS
+import qualified Network.IPFS.Process.Error as Process
 
 class Monad m => MonadLocalIPFS m where
-  withIPFSProc ::
-               (ProcessConfig stdin stdout () -> m a)
-               -> StreamIn  stdin
-               -> StreamOut stdout
-               -> [Opt]
-               -> m a
-  ipfsRun  :: [Opt] -> Lazy.ByteString -> m (ExitCode, Lazy.ByteString, Lazy.ByteString)
-  getTimeout :: m IPFS.Timeout
+  ipfsRun ::
+    [Opt]
+    -> Lazy.ByteString
+    -> m (Either Process.Error Process.RawMessage)
```

Comment:
💅

I'd align the types directly:

```haskell
class Monad m => MonadLocalIPFS m where
  ipfsRun ::
       [Opt]
    -> Lazy.ByteString
    -> m (Either Process.Error Process.RawMessage)
```

Thoughts?

---

Also, for readability have you thought about `runIPFS` or `withIPFS`? (Or with qualified imports: `IPFS.run`)

Reply:
Sent you a message on ^^ but just to make it canonical: 
I haven't been aligning types in any of my type signatures
ie 
```
rawList ::
  MonadLocalIPFS m
  => m (Either Process.Error Process.RawMessage)
rawList = ipfsRun ["bootstrap", "list"] ""
```
granted nearly all of them start with constraints and I think that makes it less unclear than with types.

Reply:
Good suggestion^^ I changed it to `runLocal` to mirror the Remote class and then used qualified imports so it reads `IPFS.runLocal`